src/sinks/datadog/logs/service.rs

```diff
@@ -144,12 +150,14 @@ impl HttpSink for Service {
             .header("Content-Type", "application/json")
             .header("DD-API-KEY", &api_key[..]);
 
-        let (request, body) = match self.compression {
+        let (request, encoded_body) = match self.compression {
             Compression::None => (request, body),
             Compression::Gzip(level) => {
                 let level = level.unwrap_or(GZIP_FAST);
-                let mut encoder =
-                    GzEncoder::new(Vec::new(), flate2::Compression::new(level as u32));
+                let mut encoder = GzEncoder::new(
+                    Vec::with_capacity(body.len()),
```

Comment:
I think technically this should be `body.len() + OVERHEAD`, which is 10 bytes AFAICT